app/markets/ActivityTable.tsx

```diff
@@ -60,94 +97,59 @@ export const ActivityTable = ({ id }: { id: string }) => {
             <TableHead className="pr-4 text-right text-text-low-em">Date</TableHead>
           </TableRow>
         </TableHeader>
         {isLoading ? (
           <LoadingSkeleton />
         ) : (
           <TableBody className="text-base font-semibold">
-            {activities?.map(activity => {
-              const outcomes = activity.fpmm.outcomes;
-              const outcomeIndex = activity.outcomeIndex;
+            {marketTransactions?.map(transaction => {
+              const isLiquidityEvent = transaction.fpmmType === FpmmType.Liquidity;
 
-              if (!outcomes || !outcomeIndex) return null;
+              if (isLiquidityEvent)
+                return (
+                  <LiquidityEventRow
+                    key={transaction.id}
+                    transaction={transaction}
+                    aiAgentsList={aiAgentsList}
+                  />
+                );
 
-              const outcome = new Outcome(
-                outcomeIndex,
-                outcomes[outcomeIndex] ?? 'Option ' + outcomeIndex,
-                id
+              const tradeAssociatedWithTransaction = marketTrades.find(
+                trade => trade.id === transaction.id
               );
 
-              return (
-                <TableRow key={activity.transactionHash}>
-                  <TableCell className="flex items-center space-x-2 pl-4 text-text-high-em">
-                    <a
-                      href={getGnosisAddressExplorerLink(activity.creator.id)}
-                      className="hover:underline"
-                      target="_blank"
-                      rel="noopener noreferrer"
-                    >
-                      {shortenAddress(activity.creator.id)}
-                    </a>
-                    {getIsAIAgent(activity.creator.id) && (
-                      <Image src="/ai.svg" alt="ai" width={18} height={18} />
-                    )}
-                  </TableCell>
-                  <TableCell>
-                    <Tag
-                      size="xs"
-                      colorScheme={TAG_COLOR_SCHEMES[outcome.index as 0 | 1]}
-                      className="w-fit uppercase"
-                    >
-                      {outcome.symbol}
-                    </Tag>
-                  </TableCell>
-                  <TableCell className="truncate text-text-high-em">
-                    {formatEtherWithFixedDecimals(activity.outcomeTokensTraded)}
-                  </TableCell>
-                  <TableCell className="text-nowrap pr-4 text-right text-text-low-em">
-                    {formatDateTime(activity.creationTimestamp)}
-                  </TableCell>
-                </TableRow>
-              );
+              if (tradeAssociatedWithTransaction) {
+                const activity: MergedTradeTransaction = {
+                  ...tradeAssociatedWithTransaction,
+                  ...transaction,
+                  fpmm: tradeAssociatedWithTransaction.fpmm,
```

Comment:
If we destruct `tradeAssociatedWithTransaction` in last maybe we don't need this line

Reply:
we need, otherwise it keeps `transaction.fpmm` instead of `trade.fpmm` (and we want to keep this one).

Reply:
```suggestion
                  ...transaction,
                  ...tradeAssociatedWithTransaction,
```

If we do something like this it will keep `trade.fpmm`, tradeAssociatedWithTransaction will override transaction values. Are we doing in that order for any specific reason?